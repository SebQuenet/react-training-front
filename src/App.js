import React from 'react';

import 'scss/_main.scss';

// Redux
import { Provider } from 'react-redux';

// Routing
import { BrowserRouter } from 'react-router-dom';
import { renderRoutes } from 'react-router-config';

// Routes definitions
import routes from 'routes';

// Redux Store
import store from 'domains';

const App = () => (
  <BrowserRouter>
    <Provider store={store}>
      {renderRoutes(routes)}
    </Provider>
  </BrowserRouter>
);

export default App;
