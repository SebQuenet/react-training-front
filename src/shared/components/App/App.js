// @flow

import React, { useEffect, useState } from 'react';
import axios from 'axios';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Books from '../Books/Books';

import './App.css';

/**
 * App component
 */
const App = () => {
  const [books, setBooks] = useState([]);

  /**
   * retrieve all GOT books
   * @returns {Promise<void>}
   */
  async function fetchBookData() {
    const { data } = await axios.get('https://anapioficeandfire.com/api/books');
    setBooks(data);
  }

  useEffect(() => {
    fetchBookData();
  });

  return (
    <div className="App">
      <Grid container spacing={3}>
        <Grid item xs={12}>
          <Typography className="App-Typography" variant="h4" component="h2" gutterBottom>
            ABBEAL - React Training Front Perfomed By Beauclair BILONG
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Books books={books} />
        </Grid>
      </Grid>
    </div>
  );
};

export default App;
