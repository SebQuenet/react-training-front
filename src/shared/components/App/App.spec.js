import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

jest.mock('../Books/Books', () => 'mock-books-component');

describe('App component', () => {
  it('should render correctly', () => {
    // given
    const component = renderer.create(<App />);
    // when
    const tree = component.toJSON();
    // then
    expect(tree).toMatchSnapshot();
  });
});
