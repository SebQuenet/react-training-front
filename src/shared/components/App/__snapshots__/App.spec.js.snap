// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<div
  className="App"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        className="MuiTypography-root App-Typography MuiTypography-h4 MuiTypography-gutterBottom"
      >
        ABBEAL - React Training Front Perfomed By Beauclair BILONG
      </h2>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <mock-books-component
        books={Array []}
      />
    </div>
  </div>
</div>
`;
